lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -627,6 +628,19 @@ protected function prepareUpdateData($entity)
                 $fieldMapping = $this->class->fieldMappings[$field];
                 $columnName   = $fieldMapping['columnName'];
 
+                $isInsert = ! empty($this->queuedInserts[spl_object_id($entity)]);
+                if (! $isInsert && ! $this->class->isUpdateable($field)) {
```

Comment:
You don't need this, you already have `$fieldMapping` in 628 and can just test for `$fieldMapping['updateable']`.